install-linux.sh

```diff
@@ -0,0 +1,162 @@
+#!/bin/bash
+
+# Parse command-line arguments for the --no-docker flag
+for arg in "$@"
+do
+    case $arg in
+        --no-docker)
+        NO_DOCKER=1
+        shift # Remove --no-docker from processing
+        ;;
+        *)
+        # Unknown option
+        ;;
+    esac
+done
+
+# Function to check if Docker is installed
+check_docker() {
+    echo "Checking for Docker..."
+    if ! command -v docker &> /dev/null; then
+        echo "Docker could not be found. Would you like to install Docker? (y/n)"
+        read -r choice
+        if [ "$choice" = "y" ]; then
+            install_docker
+        else
+            echo "Docker is required for the Docker-based installation. Proceeding with Docker-less installation."
+            dockerless_install
+            exit 0
+        fi
+    else
+        echo "Docker is already installed."
+    fi
+}
+
+# Function to install Docker
+install_docker() {
+    echo "Installing Docker..."
+    sudo apt-get update
+    sudo apt-get install -y \
+        ca-certificates \
+        curl \
+        gnupg \
+        lsb-release
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+    echo \
+      "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+      $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+    sudo apt-get update
+    sudo apt-get install -y docker-ce docker-ce-cli containerd.io
+    sudo docker run --rm hello-world
+    if [ $? -ne 0 ]; then
+        echo "Docker installation failed. Please check the error messages above."
+        exit 1
+    fi
+    echo "Docker has been successfully installed."
```

Comment:
If we use `https://get.docker.com/` we can probably support other distro.

Reply:
Ooh nice, didn't know this existed. I wanted to do the same with Ollama's installer script too, so may as well do both this way and not have to maintain more code than necessary.